zebrad/src/components/mempool/error.rs

```diff
@@ -0,0 +1,29 @@
+//! Errors that can occur when adding transactions to the mempool or querying it.
+
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+#[allow(dead_code)]
+pub enum TransactionInsertError {
```

Comment:
Can the mempool update/delete transactions ?

If so, are we going to add possible errors into this enum ?

Reply:
Definitely the mempool will drop transactions. If we drop by hash then it can fail if it is not found, if we just delete the first one the only failure will be if the mempool is empty.

Reply:
I am just writing ideas that are coming to my mind for discussion, please feel free to ignore them.

What about a nested error structure as:

```rust
enum MempoolError {
    Insert,
    Drop,
    Query,
};
enum Insert {
   AlreadyExist,
   Invalid,
   ...
}
```

So, when we use them we can do stuff like `return Err(MempoolError::Insert::AlreadyExist)`.

Reply:
> What about a nested error structure

Rust enums don't quite work like this, so we usually do:

```
enum MempoolError {
    Insert(MempoolInsertError),
    Drop(MempoolDropError),
    Query(MempoolQueryError),
};
enum MempoolInsertError {
   AlreadyExist,
   Invalid,
   ...
}
impl From<MempoolInsertError> for MempoolError { ... }
```

And end up with code like `return Err(MempoolInsertError::AlreadyExist.into()).`

I think this change is optional.

Reply:
Maybe Blocker

> Can the mempool update/delete transactions ?
> 
> If so, are we going to add possible errors into this enum ?

I also asked a similar question at:
https://github.com/ZcashFoundation/zebra/pull/2615#issuecomment-898055391

> I'm wondering how we want to handle transactions that get dropped after they are inserted into the mempool.
> (For example, transactions that are too old, or a mempool that is too big.)
> 
> There's some overlap with TransactionInsertError variants, so we might want to work that out as part of this PR.
> (For example, if a block containing a transaction is committed after the transaction enters the mempool, we want to drop that transaction.)

Since the mempool service is blocking on this PR, here's what I suggest:
1. Use a single `MempoolError` type in this PR
2. Open a follow-up ticket to split the errors into categories, once we know what all the errors are